### PR TITLE
Log more to InternalLogger

### PR DESCRIPTION
**Pull request by 304NotModified**
fixes https://github.com/NLog/NLog/issues/1192

### CODE REVIEW

**Comment by 304NotModified** [comment]
deadlocks, see

https://github.com/NLog/NLog/issues/1218

File: src/NLog/Targets/Wrappers/AsyncTargetWrapper.cs [hunk 1]
```diff
@@ -255,14 +257,21 @@ private void ProcessPendingEvents(object state)
 
             try
             {
+
+                if (this.WrappedTarget == null)
+                {
+                    InternalLogger.Error("AsyncWrapper: WrappedTarget is NULL");
+                    return;
```
Comment (304NotModified) [inline src/NLog/Targets/Wrappers/AsyncTargetWrapper.cs]: this is bad I guess, (deadlock)

File: src/NLog/Targets/Wrappers/BufferingTargetWrapper.cs [hunk 2]
```diff
@@ -130,6 +130,7 @@ protected override void FlushAsync(AsyncContinuation asyncContinuation)
             }
             else
             {
+                InternalLogger.Trace("BufferingWrapper: Flush {0} events async", events.Length);
```
Comment (304NotModified) [inline src/NLog/Targets/Wrappers/BufferingTargetWrapper.cs]: todo check events for `null`

**Comment by 304NotModified** [comment]
deadlock from [`InternalLogger.LogToConsole = true;` ](https://github.com/NLog/NLog/blob/19f471d7dfdd2bf6455b347f885ecff2cf22bf2e/tests/NLog.UnitTests/Targets/ConcurrentFileTargetTests.cs#L91). Dunno why

**Comment by 304NotModified** [comment]
@bhaeussermann / @Page-Not-Found any idea why this change leads to a deadlock in  NLog.UnitTests.Targets.ConcurrentFileTargetTests.BufferedConcurrentTest (also local)?

**Comment by bhaeussermann** [comment]
@304NotModified I can't figure it out either. I will have a closer look when I can make the time.

**Comment by 304NotModified** [comment]
I found out it has something to do with the console output of the internal logger.

**Comment by 304NotModified** [comment]
Xamarin again..

**Pull request merged**